spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -599,46 +602,15 @@ private static void deleteManifests(FileIO io, List<ManifestFile> manifests) {
         .run(item -> io.deleteFile(item.path()));
   }
 
-  // Attempt to use Spark3 Catalog resolution if available on the path
-  private static final DynMethods.UnboundMethod LOAD_METADATA_TABLE = DynMethods.builder("loadMetadataTable")
-      .hiddenImpl("org.apache.iceberg.spark.Spark3Util", SparkSession.class, Table.class, MetadataTableType.class)
-      .orNoop()
-      .build();
-
-  public static Dataset<Row> loadCatalogMetadataTable(SparkSession spark, Table table, MetadataTableType type) {
```

Comment:
Just flagging that this will remove public method, in case anyone may use this method (though I feel there should not be).

Reply:
Good point, I missed it. I wonder why we made it public, though. I'd probably remove the method but I don't mind deprecating and delegating to the one below if folks think this may impact anyone.

Reply:
Yeah, this is a grey area, in which these methods actually are APIs, but not considered as APIs.  Per our community sync, only the public things in the api module are considered as APIs. We need something(e.g. annotation) to mark these methods as APIs.

Reply:
Sounds good to me.